### DISCUSSION TITLE
Cannot find module 'vue-tsc/out/index'

### DISCUSSION THREAD

**Discussion by Reconnact** [nuxt/nuxt#25530] [General]
I'm trying to get a nuxt application to work with nuxt typescript typechecking: https://nuxt.com/docs/guide/concepts/typescript

For some reason I always get the following error:
 
``` 
 ERROR  Cannot start nuxt:  Cannot find module 'vue-tsc/out/index'                                 2:17:26 PM
Require stack:
- /Users/username/node_modules/vite-plugin-checker/dist/esm/checkers/vueTsc/prepareVueTsc.js

  Require stack:
  - /Users/username/node_modules/vite-plugin-checker/dist/esm/checkers/vueTsc/prepareVueTsc.js
  at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
  at Function.resolve (node:internal/modules/helpers:187:19)
  at <anonymous> (/Users/username/node_modules/vite-plugin-checker/src/checkers/vueTsc/prepareVueTsc.ts:14:31)
  at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
  at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
  at createCheckers (/Users/username/node_modules/vite-plugin-checker/src/main.ts:48:37)
  at config (/Users/username/node_modules/vite-plugin-checker/src/main.ts:88:18)
  at async runConfigHook (/Users/username/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64543:25)
  at async resolveConfig (/Users/username/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64026:14)
  at async _createServer (/Users/username/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:63274:20)
  ```

Environment:
```
------------------------------
- Operating System: Darwin
- Node Version:     v21.4.0
- Nuxt Version:     3.6.5
- CLI Version:      3.10.0
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.22.21
- Builder:          -
- User Config:      app, css, components, plugins, imports, modules, i18n, build, postcss, matomo, runtimeConfig, vite, dev, typescript
- Runtime Modules:  @nuxt/content@2.11.0, @nuxtjs/tailwindcss@6.11.2, @nuxtjs/i18n@8.0.0, @vueuse/nuxt@10.7.2, ./modules/matomo
- Build Modules:    -
------------------------------ 
```

Note: I didn't create a bugreport as that seems to work for others

**Comment by ukramer** [comment]
For me it helped reinstalling the dependencies with the package manager that I chose during set up of the project, e.g. `pnpm add -D vue-tsc typescript`

**Reply by Reconnact** [reply]
Hmm. I did this several times now (including removing node-modules, yarn.lock file, etc.) and I keep getting the same error. I also checked if the module exists in my node modules and they do. I'm thinking this is more of an issue that the access is somehow restricted or anything. I also have a weird issue with my local setup when running a nuxt project without setting:
```
vite: {
    server: {
        fs: {
          strict: false
        }
    }
},
 ```
Here you can see more: https://stackoverflow.com/questions/74902697/error-the-request-url-is-outside-of-vite-serving-allow-list-after-git-init

Is it possible that is related to that? I'm thinking about completely wiping evrything on my machine and reinstalling but I would like to avoid that

**Comment by mitjans** [comment]
I just had the same problem after upgrading [vue-tsc](https://github.com/vuejs/language-tools) to the latest `^2.0.0` version. I still haven't found a way of fixing it.

**Comment by sullyD64** [comment]
Probably related to a breaking change in vue-tsc v2

https://github.com/vuejs/language-tools/issues/3969

**Comment by octo-kumo** [comment]
I was able to solve this issue by removing type check.
```ts
export default defineNuxtConfig({
    typescript: {
        // typeCheck: true
    }
}
```

**Reply by carsonbird** [reply]
Does that... Disable type checking? What if I want type checking, at least in my IDE while developing. Does this just disable a final type check during generation time?

**Reply by gmmortera** [reply]
Do you have any idea why this action fix the issue?

**Comment by dsvgl** [comment]
Well known error. Stick to vue-tsc@1
See https://nuxt.com/docs/guide/concepts/typescript#type-checking

**Comment by digitalniweb** [comment]
Looks like finally fixed
"nuxt": "^3.12.3",
"vue-tsc": "^2.0.24"

**Comment by LorenzoRottigni** [comment]
Hello there, i'm receiving same errors...
is vue-tsc 2.0.24 instead of 2.0.26 the solution?

Im running nuxt 3.11.2 and vue-tsc 2.0.26 and definitely it isn't working...
Here my tsconfig contained in nuxt.config.ts:

typescript: {
    strict: true,
    typeCheck: true,
    tsConfig: {
      compilerOptions: {
        // necessary for apollo-upload-client
        allowJs: true,
        maxNodeModuleJsDepth: 10,
        module: 'NodeNext',
        moduleResolution: 'NodeNext',
        esModuleInterop: true,
        resolveJsonModule: true,
        types: [
          '@pinia/nuxt',
          '@nuxtjs/i18n',
          '@types/node',
          '@nuxt/image',
          '@nuxt/types',
          "@types/apollo-upload-client"
        ]
      }
    }
  }

**Reply by xoxefdp** [reply]
Go for nuxt@3.12.3, its solved there

**Comment by LorenzoRottigni** [comment]
Furthermore package @types/apollo-upload-client isn't working well, adding it to compilerOptions.types doesn't augment typescript types.... I think that package apollo-upload-client has a too wide compatibility range... resulting in a messy setup/integration.
Why it doesn't expose a js dist with entrypoints specified inside package.json!? that's not the standard for publishing packages.


```
 ERROR                                                                   5:27:02 PM
 ERROR(vue-tsc)  Cannot find type definition file for '@types/apollo-upload-client'.
  The file is in the program because:
    Entry point of type library '@types/apollo-upload-client' specified in compilerOptions
```

I was correctly using package apollo-upload-client on nuxt2 but I can't find a way to make it working on nuxt3.

Trick from Google Gemini (still to be tried):
`import createUploadLink from 'apollo-upload-client/createUploadLink.mjs';`

**Comment by richardevcom** [comment]
Had the same error when building my Nuxt `3.14.1592` app.
**:point_right: Fixed it by adding `typescript`: `~5.6.2` and `vue-tsc`: `2.1.10`** _(checked needed versions in `vue` and `nuxt` `package.json`)_ **as dev dependencies in my `package.json`.**
